### PR TITLE
refactored backend for easier configuration, Added docstrings & applied better debug messages

### PR DESCRIPTION
**Pull request by Blankscreen-exe**
Here's what I did:

- Refactored backend code for better configuration(inspired by django)
    - added **settings.py** in root directory (store all your configurations and .env data in it)
    - added **.env.example** which contains basic secrets
    - added **routes/routes.py** for route constants to be used in **main.py**
- Added proper docstrings to most of the classes, methods and functions
- Added system for better debug messages✦

 ✦ **terminal_colors.py** class contains color configurations for stdout messages. I used them in debug messages all over backend and made them runnable only when `DEBUG` in **.env** is set to `True`

### CODE REVIEW

File: backend/tests/terminal_colors.py
Comment (grumpyp) [inline backend/tests/terminal_colors.py:1]: This is not a test. I havn't seen that usage either anywhere else.

Comment (Blankscreen-exe) [reply backend/tests/terminal_colors.py:1]: it is just for better debug messages, which can easily be distinguishable from other api requests. Also those logs wont show if DEBUG is set to true. I can remove it if you dont want that

**Comment by grumpyp** [comment]
Hi @Blankscreen-exe whats the status here?

**Comment by Blankscreen-exe** [comment]
doing a fresh start here

**Pull request closed without merging**